roles/openshift_docker/tasks/main.yml

```diff
@@ -0,0 +1,28 @@
+---
+# It's important that we don't explicitly pull this image here.  Otherwise we
+# could result in upgrading a preinstalled environment.  We'll have to set
+# openshift_image_tag correctly for upgrades.
+- name: Set version when containerized
+  command: >
+    docker run --rm {{ openshift.common.cli_image }}:latest version
+  register: cli_image_version
+  when: openshift.common.is_containerized | bool and openshift_image_tag is not defined
```

Comment:
Shouldn't this value be set in openshift_facts and use the facts here? I hate to rely just on passed in values.